packages/wdio-repl/src/index.ts

```diff
@@ -73,71 +81,79 @@ export default class WDIORepl {
-            return callback(e)
+            this._isCommandRunning = false
+            return callback(e, undefined)
         }
     }
 
-    _handleResult(result, callback) {
+    private _handleResult (result: any, callback: (err: Error | null, result: any) => void) {
         if (!result || typeof result.then !== 'function') {
-            this.isCommandRunning = false
+            this._isCommandRunning = false
             return callback(null, result)
         }
 
+        let timeoutCalled = false
         const timeout = setTimeout(
             () => {
-                callback(new Error('Command execution timed out'))
-                this.isCommandRunning = false
-                timeout._called = true
+                callback(new Error('Command execution timed out'), undefined)
+                this._isCommandRunning = false
+                timeoutCalled = true
             },
-            this.config.commandTimeout
+            this._config.commandTimeout
         )
 
-        result.then((res) => {
+        result.then((res: any) => {
             /**
              * don't do anything if timeout was called
              */
-            if (timeout._called) {
+            if (timeoutCalled) {
                 return
             }
-            this.isCommandRunning = false
+            this._isCommandRunning = false
             clearTimeout(timeout)
             return callback(null, res)
-        }, (e) => {
+        }, (e: Error) => {
             /**
              * don't do anything if timeout was called
              */
-            if (timeout._called) {
+            if (timeoutCalled) {
                 return
             }
 
-            this.isCommandRunning = false
+            this._isCommandRunning = false
             clearTimeout(timeout)
             const errorMessage = e ? e.message : 'Command execution timed out'
             const commandError = new Error(errorMessage)
             delete commandError.stack
-            return callback(commandError)
+            return callback(commandError, undefined)
         })
     }
 
-    start(context) {
-        if (this.replServer) {
+    start (context: vm.Context) {
+        if (this._replServer) {
             throw new Error('a repl was already initialised')
         }
 
         if (context) {
-            const evalFn = this.config.eval
-            this.config.eval = (cmd, _, filename, callback) => evalFn(cmd, context, filename, callback)
+            const evalFn = this._config.eval
+            this._config.eval = function (cmd, _, filename, callback) {
+                return evalFn.call(this, cmd, context, filename, callback)
+            }
         }
 
-        this.replServer = repl.start(this.config)
+        this._replServer = repl.start(this._config)
 
-        return new Promise(
-            (resolve) => this.replServer.on('exit', resolve))
+        return new Promise((resolve) => {
+            if (!this._replServer) {
```

Comment:
yeah, it can't be defined at initialisation because the interface only starts it when the start method is being executed.

Reply:
Usually this is not necessary but TypeScript was not happy without checking

Reply:
This is because replServer can be undefined according to line 53.
Anyway, it's also possible to use 'as'
(this._replServer as ReplServerTypeHere).on(...